src/main/kotlin/io/ionic/libs/ionfilesystemlib/helper/common/IONFILEUriResolveExtensions.kt

```diff
@@ -0,0 +1,67 @@
+package io.ionic.libs.ionfilesystemlib.helper.common
+
+import io.ionic.libs.ionfilesystemlib.helper.IONFILEUriHelper
+import io.ionic.libs.ionfilesystemlib.model.IONFILEExceptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEUri
+import io.ionic.libs.ionfilesystemlib.model.LocalUriType
+
+internal suspend inline fun <reified T> IONFILEUriHelper.useUriIfResolvedAsLocalDirectory(
+    uri: IONFILEUri,
+    onResolved: (IONFILEUri.Resolved.Local) -> Result<T>
+): Result<T> = useUriIfResolvedAsLocal(uri) { resolvedUri ->
+    if (resolvedUri.type != LocalUriType.FILE) {
```

Comment:
Just to make sure I understand, why don't we do `resolvedUri.type == LocalUriType.DIRECTORY)` instead?

Reply:
Oh that makes sense. In that case I would just add a comment explaining it yeah.